server/api/impl/upstream/upawslambda/upstream.go

```diff
@@ -27,14 +42,52 @@ func NewUpstream(app *apps.App, aws *aws.Client) *Upstream {
 }
 
 func (u *Upstream) OneWay(call *apps.Call) error {
-	_, err := u.aws.InvokeLambda(u.app.Manifest.AppID, u.app.Manifest.Version, call.URL, lambda.InvocationTypeEvent, call)
+	payload, err := CallToInvocationPayload(call)
+	if err != nil {
+		return errors.Wrap(err, "failed to covert call into invocation payload")
+	}
+
+	_, err = u.aws.InvokeLambda(u.app.Manifest.AppID, u.app.Manifest.Version, u.app.Manifest.Functions[0].Name, lambda.InvocationTypeEvent, payload)
 	return err
 }
 
 func (u *Upstream) Roundtrip(call *apps.Call) (io.ReadCloser, error) {
-	bb, err := u.aws.InvokeLambda(u.app.Manifest.AppID, u.app.Manifest.Version, call.URL, lambda.InvocationTypeRequestResponse, call)
+	payload, err := CallToInvocationPayload(call)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to covert call into invocation payload")
+	}
+
+	bb, err := u.aws.InvokeLambda(u.app.Manifest.AppID, u.app.Manifest.Version, u.app.Manifest.Functions[0].Name, lambda.InvocationTypeRequestResponse, payload)
 	if err != nil {
 		return nil, err
 	}
-	return ioutil.NopCloser(bytes.NewReader(bb)), err
+
+	var resp invocationResponse
+
+	err = json.Unmarshal(bb, &resp)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error marshaling request payload")
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.Errorf("lambda invocation failed with status code %v and body %v", resp.StatusCode, resp.Body)
+	}
+
+	return ioutil.NopCloser(strings.NewReader(resp.Body)), nil
+}
+
+func CallToInvocationPayload(call *apps.Call) ([]byte, error) {
+	request := invocationPayload{
```

Comment:
What is the reason of having the HTTP like payload? Matching with the HTTP apps?

Reply:
Apps expect an HTTP like payload given that they are essentially HTTP servers. That approach makes the transition from an HTTP app to an AWS app seamless. Given that we don't place an AWS Gateway in front of the lambda we have to generate the payload ourself.

Reply:
I 👍  @hanzei's view here. This approach would also make a transition to a self-hosted (lambda) easier on the developer, should they choose to do so.